src/backend/web.rs

```diff
@@ -962,6 +962,16 @@ impl crate::Context for Context {
         wgt::Limits::default()
     }
 
+    fn device_get_swap_chain_preferred_format(
+        &self,
+        device: &Self::DeviceId,
+    ) -> wgt::TextureFormat {
+        let device_promise = surface.0.get_swap_chain_preferred_format(&device.0);
```

Comment:
This doesn't need to be a promise. We can for now leave it as `todo!` until the web-sys bindings are updated with it being non-promise.

Reply:
Should we make it return the default texture format (`Bgra8UnormSrgb` according to [wgpu/src/device/mod.rs](https://github.com/gfx-rs/wgpu/blob/2287ae3f8a7246e8b47b73a1de6fe2410c42b8c8/wgpu-core/src/device/mod.rs#L3853)) like this:
```rust
fn device_get_swap_chain_preferred_format(
    &self,
    device: &Self::DeviceId,
) -> wgt::TextureFormat {
    // TODO: web-sys bindings need to be updated to not return a promise
    wgt::TextureFormat::Bgra8UnormSrgb
}
```

Or just leave it as `todo!` for future work?

Reply:
Resolved on https://github.com/gfx-rs/wgpu-rs/pull/693/commits/45dd8e17b4ca06b68ce45dfdad911113ecd2fb44